src/store/reducers/cartReducer.jsx

```diff
@@ -0,0 +1,53 @@
+import { createSlice } from '@reduxjs/toolkit';
+
+const initialState = {
+  list: [],
+};
+
+export const cartSlice = createSlice({
+  name: 'cart',
+  initialState,
+  reducers: {
+    add: (state, action) => {
+      // state.list = [...state.list, action.payload];
+
+      let existingItem = state.list.find(
+        (item) => item.id === action.payload.id
+      );
+
+      if (existingItem) {
+        return {
+          ...state,
+          list: state.list.map((dish) => {
+            return dish.id === action.payload.id
+              ? {
+                  ...dish,
+                  quantity: dish.quantity + action.payload.quantity,
+                }
+              : dish;
+          }),
+        };
+      }
+      return { ...state, list: [...state.list, action.payload] };
+    },
+    changeAmount: (state, action) => {
+      state.list = state.list
```

Comment:
afaik antipattern, you dont want to manipulate directly.
const newList = ...

return {
...state, list: newList}

Reply:
yeah, good point, noticed that myself even. fixed.